File: 4 Median of Two Sorted Arrays/4median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        nums1 += nums2
        nums1.sort()

        n = len(nums1)
        if n % 2 == 0:
            return (nums1[int(n/2) - 1] + nums1[int(n/2)])/2
        else:
            return nums1[n//2]

        # n, m = len(nums1), len(nums2)
        # if (n + m) % 2 == 0:
        #     l, u = (n + m) / 2 - 1, (n + m) / 2
        #     i = 0

        #     while n > 0 and m > 0:
        #         x, y = nums1.pop(0), nums1.pop(0)
        
        # else:
        #     l, u = (n + m) // 2, None


        

File: 4 Median of Two Sorted Arrays/4median-of-two-sorted-arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int n1 = nums1.size(), n2 = nums2.size(), i = 0, j = 0;
        int k = n1 + n2, curr = 0, prev = 0;

        for (int z = 0; z <= k / 2; z++){
            prev = curr;
            if (i < n1 && (j >= n2 || nums2[j] > nums1[i])){
                curr = nums1[i];
                i++;
            }
            else {
                curr = nums2[j];
                j++;
            }
        }

        if (k % 2 == 0) {
            return (curr + prev) / 2.0;
        }
        else {
            return curr;
        }
    }
};